components/ILIAS/Forum/classes/Tables/ForumStatisticsTable.php

```diff
@@ -0,0 +1,240 @@
+            }
+
+            private function getRecords(Range $range, Order $order): array
+            {
+                $this->initRecords();
+                $records = $this->sortedRecords($this->records, $order);
+                return $this->limitRecords($records, $range);
+            }
+
+            private function limitRecords(array $records, Range $range): array
+            {
+                return array_slice($records, $range->getStart(), $range->getLength());
+            }
+
+            public function getRows(
+                \ILIAS\UI\Component\Table\DataRowBuilder $row_builder,
+                array $visible_column_ids,
+                \ILIAS\Data\Range $range,
+                \ILIAS\Data\Order $order,
+                ?array $filter_data,
+                ?array $additional_parameters,
+            ): Generator {
+                $records = $this->getRecords($range, $order);
+                foreach ($records as $record) {
+                    $row_id = (string) $record['usr_id'];
+                    yield $row_builder->buildDataRow($row_id, $record);
+                }
+            }
+
+            public function getTotalRowCount(?array $filter_data, ?array $additional_parameters): ?int
+            {
+                $this->initRecords();
+                return count((array) $this->records);
+            }
+
+            private function getProgressStatus(int $user_id): string
+            {
+                if ($this->has_active_lp && $this->has_general_lp_access) {
+                    if ($this->has_rbac_or_position_access || $this->actor->getId() === $user_id) {
+                        switch (true) {
+                            case in_array($user_id, $this->completed, false):
+                                $status = $this->lng->txt(ilLPStatus::LP_STATUS_COMPLETED);
+                                $icon = $this->icons->renderIconForStatus(ilLPStatus::LP_STATUS_COMPLETED_NUM);
+                                break;
+
+                            case in_array($user_id, $this->in_progress, false):
+                                $status = $this->lng->txt(ilLPStatus::LP_STATUS_IN_PROGRESS);
+                                $icon = $this->icons->renderIconForStatus(ilLPStatus::LP_STATUS_IN_PROGRESS_NUM);
+                                break;
+
+                            case in_array($user_id, $this->failed, false):
+                                $status = $this->lng->txt(ilLPStatus::LP_STATUS_FAILED);
+                                $icon = $this->icons->renderIconForStatus(ilLPStatus::LP_STATUS_FAILED_NUM);
+                                break;
+
+                            default:
+                                $status = $this->lng->txt(ilLPStatus::LP_STATUS_NOT_ATTEMPTED);
+                                $icon = $this->icons->renderIconForStatus(ilLPStatus::LP_STATUS_NOT_ATTEMPTED_NUM);
+                                break;
+                        }
+                    } else {
```

Comment:
You can drop the `else` case if you define the default at the top of the method.